### PR TITLE
Astro/blog [RFR]

### PR DESCRIPTION
**Pull request by thilobillerbeck**
This project reimplements the NixOS homepage `/blog` section with omitted Community section and on page filtering of posts as well as seperate pages per post. The content has been centralized and standardized to markdown for ease of editing.

This PR does the following things: 
- add announcements, nixos-weekly and stories to the content dir and convert them to markdown if needed
- adjust code block theme to match with original site
- add article view styling to match original site
- add a blog overview with category filters (as discussed and approved in last weeks marketing call)
- add a per entry view with discrete URLs
- add RSS feed (one for everything and one for every category)
- separate image behavior in blog articles and the learn section
- adjusted header to incorporate a subheading for things like metadata

### CODE REVIEW

File: src/pages/index.astro [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+  return p.slug.split("/")?.[0] === "announcements";
+}).reverse();
 ---
 
 <Layout title="Nix & NixOS">
   <div class="bg-nixlighterblue">
     <Container class="flex flex-col md:flex-row items-center justify-between py-4">
-      <b class="text-nixsemidarkblue text-lg">NixOS 23.05 released.</b>
-      <Button>Announcement</Button>
+      <b class="text-nixsemidarkblue text-lg">{posts[0].data.title}</b>
```
Comment (djacu) [inline src/pages/index.astro:24]: this is :pinched_fingers:

File: src/components/ui/Button.astro [hunk 2]
```diff
@@ -37,11 +37,11 @@ const {
     shaText = "",
     } = Astro.props;
 
-const localClassList = `${colorMap[color]} font-bold ${sizeMap[size]} text-white drop-shadow-[0_4px_1px_rgba(0,0,0,0.2)]`;
+const localClassList = `${colorMap[color]} object-contain font-bold ${sizeMap[size]} text-white drop-shadow-[0_4px_1px_rgba(0,0,0,0.2)]`;
```
Comment (djacu) [inline src/components/ui/Button.astro:40]: Was something breaking without `object-contain`?

**Review by djacu — APPROVED** [review]
Looks really nice! I checked several of the blog entries and tried to break the filter. All seems to be working well.

side note: we should consider replacing those images on the NixOS release pages in the future. low res png doesn't look great.

File: src/components/ui/Button.astro (hunk 2 above)
Comment (thilobillerbeck) [reply src/components/ui/Button.astro]: not really, this is just a remainder of debugging. I just removed it :D

**Pull request merged**